### PR TITLE
add docs of n-queens problem

### LINKED ISSUE #1001: README for Queen's Algorithm
Add a README.md file to explain Queen's Algorithm. The file should contain the following:
- Explanation
- Algorithm
- Pseudocode
- Example
- Time Complexity

Also add images/GIFs to explain if possible.

### PR DESCRIPTION
**Pull request by vsharma20v**
Fixes #1001 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Sample Input / Output.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

#### Changes proposed in this pull request:

- add docs of n-queens problem

#### Languages Used:
- Markdown

#### Files Added:
- README.md

### CODE REVIEW

File: Queens_Problem/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# N Queens Problem
+The N Queens Problem is one of the most common examples of **backtracking**. The problem is to arrange N queens on an NxN chessboard such that no queen can strike down any other queen. A queen can attack horizontally, vertically, or diagonally.
+
+<img src="https://www.dl.dropboxusercontent.com/s/ba1ld9w66wr0qap/final.png" width="300">
+## Algorithm
```
Comment (marwahmanbir) [inline Queens_Problem/README.md]: Add a single blank line before this line.

File: Queens_Problem/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+The idea is to place queens one by one in different columns, starting from the leftmost column. When we place a queen in a column, we check for clashes with already placed queens. In the current column, if we find a row for which there is no clash, we mark this row and column as part of the solution. If we do not find such a row due to clashes then we backtrack and return false.
+- Start in the leftmost column
+- If all queens are placed return true
+- Try all rows in the current column.  Do following for every tried row.
+    - If the queen can be placed safely in this row then mark this [row, column] as part of the solution and recursively check if placing queen here leads to a solution.
+    - If placing the queen in [row, column] leads to a solution then return true.
+    - If placing queen doesn't lead to a solution then you mark this [row, column] (Backtrack) and go to step (a) to try other rows.
+- If all rows have been tried and nothing worked, return false to trigger backtracking.
+
+
```
Comment (marwahmanbir) [inline Queens_Problem/README.md]: Remove one extra line.

File: Queens_Problem/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# N Queens Problem
```
Comment (marwahmanbir) [inline Queens_Problem/README.md]: Also mention time complexity.

Comment (vsharma20v) [reply Queens_Problem/README.md]: Done

Comment (vsharma20v) [reply Queens_Problem/README.md]: @marwahmanbir plz review the changes i have made

**Pull request merged**